src/routes/+page.svelte

```diff
@@ -0,0 +1,241 @@
+	) {
+		d3.select('.tooltip')
+			.style('display', 'block')
+			.style('left', d.pageX + 'px')
+			.style('top', d.pageY - 200 + 'px')
+			.text(
+				`On ${data.date}, ${data.values['Images published with alternative text']} out of ${data.values['Images published']} images had alt text (${((data.values['Images published with alternative text'] / data.values['Images published']) * 100).toFixed(2)}%)`
+			);
+
+		d3.selectAll('.stacked-bar').style('opacity', '0.25');
+		d.target!.style.opacity = 1;
+	}
+
+	function handleMouseOut(d) {
+		if (d.toElement.nodeName !== "rect") {
+			d3.select('.tooltip').style('display', 'none');
+			d3.selectAll('.stacked-bar').style('opacity', 1);
+		}
+		
+	}
+</script>
+
+<svelte:window bind:innerHeight bind:innerWidth />
+
+<main>
+	<section>
+		<h2>Tracking The Daily's alternative text</h2>
+		<p>
+			The Daily has been tracking the number of images published with and without alternative text
+			since December 2022. The chart below shows the number of images published with and without
+			alternative text each day.
+		</p>
+		<div class="options">
+			<div class="legend">
+				<div class="legend-item">
+					<div style="background-color: lightgreen"></div>
+					<span>Images published with alternative text</span>
+				</div>
+				<div class="legend-item">
+					<div style="background-color: LightCoral"></div>
+					<span>Images published without alternative text</span>
+				</div>
+			</div>
+			<div>
+				<label for="timerange">Select timerange:</label>
+				<select bind:value={timerange}>
+					<option value={all}></option>
+					<option value={lastWeek}>Last week</option>
+					<option value={lastMonth}>Last month</option>
+					<option value={lastSixMonths}>Last six months</option>
+					<option value={lastYear}>Last year</option>
+				</select>
+			</div>
+			<div>
+				<label for="category">Select category:</label>
+				<select bind:value={category}>
+					<option value={null}>All</option>
+					<option value={46}>News</option>
+					<option value={44}>Sports</option>
+					<option value={31}>Opinion</option>
+					<option value={5}>Arts</option>
+					<option value={55}>Statement</option>
+					<option value={32}>MiC</option>
+					<option value={124}>Photos</option>
+					<option value={24}>Podcasts</option>
+					<option value={10327}>Videos</option>
+				</select>
+			</div>
+		</div>
+	</section>
+	<figure>
+		<div class="tooltip"></div>
+		<svg {width} {height}>
+			<g bind:this={gx} transform="translate({0}, {height - padding})" />
+			<g bind:this={gy} transform="translate({padding}, {0})" />
+			<g>
+				{#each index as [date, values]}
+					<!-- svelte-ignore a11y-no-static-element-interactions -->
+					<!-- svelte-ignore a11y-mouse-events-have-key-events -->
+					<g
+						class="stacked-bar"
+						style="margin: 0; padding; 0; gap: 0;"
+						on:mouseenter={(d) => handleMouseOver(d, { date, values })}
+						on:mouseout={(d) => handleMouseOut(d)}
+					>
```

Comment:
Changing the event handler from `mouseout` to `mouseleave`, and adding a `role="none"` will allow us to remove the accessibility warnings. A `mouseout` handler expects a corresponding `blur` handler.

According to the [SVG ARIA Semantics](https://svgwg.org/svg2-draft/struct.html#implicit-aria-semantics), a `g` element will implicitly have a `group` role if it meets the inclusion criteria. Otherwise, it will have a `none` role.

The inclusion criteria is described in the following:

> For authors, the preferred means of indicating semantic importance is to provide an accessible name for the element. This can be done through a direct child ‘[title](https://svgwg.org/svg2-draft/struct.html#TitleElement)’ element, or through the ‘[aria-label](https://www.w3.org/TR/wai-aria-1.1/#aria-label)’ or ‘[aria-labelledby](https://www.w3.org/TR/wai-aria-1.1/#aria-labelledby)’ attributes.

We don't currently meet the inclusion criteria so the role is already `none`, but the linter is not so comprehensive to know this. So, we should set it explicitly.

Even before adding a `none` role, the screen reader does not comprehend the content of the `g` elements. However, note that the screen reader does read the axes labels, which is not ideal because there is no other context about what the numbers even mean.

Here are a few articles that go over different methods of improving data visualization accessibility (in the context of election maps):

<https://fossheim.io/writing/posts/accessible-dataviz-us-elections/>
<https://washpost.engineering/improving-screen-reader-and-keyboard-accessibility-on-election-results-pages-e6f6cdbab25f>

For simplicity, I think the best approach at the moment is to provide an overall summary for the SVG element, while leaving the individual `g` elements hidden from the screen reader:

> Since the exact numbers aren’t as important here, VoiceOver doesn’t read or link to them and instead reads a summary of what’s displayed.

Sara Fossheim's article shows an example of how to do that.

```suggestion
					<g
						class="stacked-bar"
						role="none"
						style="margin: 0; padding; 0; gap: 0;"
						on:mouseenter={(d) => handleMouseOver(d, { date, values })}
						on:mouseleave={(d) => handleMouseOut(d)}
					>
```